tests/unit/NetworkTest.js

```diff
@@ -209,3 +221,45 @@ test('consecutive API calls eventually succeed when authToken is expired', () =>
             expect(chatList).toEqual(TEST_CHAT_LIST);
         });
 });
+
+test('retry network request if auth and credentials are not read from Onyx yet', () => {
+    Onyx.addDelayToConnectCallback(3000);
+
+    // Reset isReady for this unit test, because Onyx.clear() in beforeEach resolves the
+    // Onyx.connect callbacks with null values
+    Network.setIsReady(false);
+
+    // Set values to trigger Onyx.connect callbacks
+    Onyx.merge(ONYXKEYS.CREDENTIALS, {});
+    Onyx.merge(ONYXKEYS.SESSION, {});
+
+    // Functions to mock and that we want to track
+    const spyNetworkSetIsReady = jest.spyOn(Network, 'setIsReady');
+    const spyHttpUtilsXhr = jest.spyOn(HttpUtils, 'xhr').mockImplementation(() => Promise.resolve({}));
+
+    // Given a test user login and account ID
+    const TEST_USER_LOGIN = 'test@testguy.com';
+    fetchAccountDetails(TEST_USER_LOGIN);
+
+    // Wait for the mock Onyx.callbacks to be set
+    return waitForPromisesToResolve().then(() => {
+        // We should expect not having the Network ready and not making an http request
+        expect(spyNetworkSetIsReady).not.toHaveBeenCalled();
+        expect(spyHttpUtilsXhr).not.toHaveBeenCalled();
+
+        // Resolve Onyx.connect callbacks
+        jest.runOnlyPendingTimers();
+
+        // We should expect call Network.setIsReady(true)
+        expect(spyNetworkSetIsReady).toHaveBeenLastCalledWith(true);
+
+        // We should expect not making an http request yet
+        expect(spyHttpUtilsXhr).not.toHaveBeenCalled();
+
+        // Run processNetworkRequestQueue in the setInterval of Network.js
+        jest.runOnlyPendingTimers();
```

Comment:
Same question here - since if we want to advance to get the Network loop to tick then it's 1 second (and we could also set up a constant for that).

Reply:
Sure, I will change it to `jest.advanceTimersByTime(NETWORK_INTERVAL_MS);` where `NETWORK_INTERVAL_MS = 1000`